### ISSUE TITLE
Nightly rustdoc warns about code block if it contains comments and tab

### ISSUE THREAD

**Issue by pravic** [rust-lang/rust#48068]
A code block inside documentation with `rust,no_run` directive:

```Rust
let mut p = String::from("/path");
p.clear();	/* some comment indented with tab */
```

Nightly compiler throws a warning about it:

>    Doc-tests hello
WARNING: src\lib.rs -  (line 5) Code block is not currently run as a test, but will in future versions of rustdoc. Please ensure this code block is a runnable test, or use the `ignore` directive.
running 0 tests
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

And `ignore` doesn't help as well though.

Full sample: https://gist.github.com/anonymous/56d5eb5927c50b2a3a22f67723363f45#file-playground-rs-L5-L8

Rust version:

> rustc 1.25.0-nightly (bd98fe0c0 2018-02-06)
binary: rustc
commit-hash: bd98fe0c05601d8a318a709adc4263f3348ea8d4
commit-date: 2018-02-06
host: i686-pc-windows-msvc
release: 1.25.0-nightly
LLVM version: 4.0

**Comment by kennytm** [comment]
It is warned starting on nightly only because of #47398. If you `--enable-commonmark` the warnings will be emitted on 1.23.0 (stable) as well. No repro on 1.22.1 though.

cc @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Looking what's wrong.

**Comment by GuillaumeGomez** [comment]
The warning comes if you have the following code:

    /*! Sample documentation.
    Just a library with some code:
    ```rust,no_run
    let mut p = String::from("/path");
    p.clear();	/* some comment */
    ```
    */

But not if you have:

    /*! Sample documentation.
    
    Just a library with some code:
    
    ```rust,no_run
    let mut p = String::from("/path");
    p.clear();	/* some comment */
    ```
    */

The whole issue being that hoedown doesn't recognize the first one as being a code block. I'll close this issue for now. This won't be thrown out when the new release will get out anyway since all hoedown will be removed from the compiler.

**Comment by pravic** [comment]
@GuillaumeGomez But the gist contained space between comment and code block?

**Comment by steveklabnik** [comment]
yes, re-opening since the gist does look like what is supposed to work.

**Comment by ollie27** [comment]
The issue here is that hoedown converts tabs to spaces in code blocks but pulldown-cmark doesn't so rustdoc thinks it has found a new test.

**Comment by GuillaumeGomez** [comment]
Since we don't have hoedown around anymore, can we close this issue?

**Comment by pravic** [comment]
At the moment there is no warning neither in stable nor nightly :)

**Comment by ollie27** [comment]
The issue will only affect 1.25, so it's probably too late to fix this now.

**Comment by GuillaumeGomez** [comment]
Then I close it for good!
